### DISCUSSION TITLE
can I send a custom http header with wdio?

### DISCUSSION THREAD

**Discussion by RyanRedpath** [webdriverio/webdriverio#8030] [Community]
I am looking to resolve an issue we have with our test environment by creating a special "header" that our api can use to disable certain barriers we have within our test environment that causes our tests to be flakey. 

What I need to do within the tests I write, is send a custom http header. Is this possible with WDIO test framework?

**Comment by christian-bromann** [comment]
You can check `navigator.webdriver` which should be `true` when you automate a browser through WebDriver. If the flag is set you can have your application send that special header. Another option is to use CDP and [Puppeteer](https://webdriver.io/docs/api/browser/getPuppeteer) to add headers to requests via CDP.

**Reply by RyanRedpath** [reply]
Thanks @christian-bromann for the suggestions. Our resolution in the end was to utilise auth0 management api client, from this we could set a flag to disable single session for the app user account and enable in the config after session to minimise risk. 

https://github.com/auth0/node-auth0#management-api-client
